### PR TITLE
imath 3.1.9

### PR DESCRIPTION
**Pull request by miccal**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.<details>
  <summary>release notes</summary>
  <pre>Patch release that addresses miscellaneous minor compiler/build/doc issues and extends test coverage, including:

#317 fix the macro switching half-precision floating-point format implementation.
#314 Remove duplicate IMATH_HOSTDEVICE
#313 IMATH_EXPORT for Rand32::nextf()

See [CHANGES.md](CHANGES.md) for complete details.

</pre>
</details>

### CODE REVIEW

**Comment by carlocab** [comment]
`gstreamer` needs the self-hosted runner

**Comment by carlocab** [comment]
GStreamer also needs this patch to silence linkage warnings:
```diff
diff --git a/Formula/gstreamer.rb b/Formula/gstreamer.rb
index b5bb27e562b..422fb328b5a 100644
--- a/Formula/gstreamer.rb
+++ b/Formula/gstreamer.rb
@@ -173,8 +173,10 @@ class Gstreamer < Formula
     # Set `RPATH` since `cargo-c` doesn't seem to.
     # https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/issues/279
     plugin_dir = lib/"gstreamer-1.0"
-    rpath_args = [loader_path, rpath(source: plugin_dir)].map { |path| "-rpath,#{path}" }
-    ENV["RUSTFLAGS"] = "--codegen link-args=-Wl,#{rpath_args.join(",")}"
+    rpath_arg = [loader_path, rpath(source: plugin_dir)].map { |path| "-rpath,#{path}" }
+                                                        .join(",")
+    ENV["RUSTFLAGS"] = "--codegen link-args=-Wl,#{rpath_arg}"
+    ENV.append "LDFLAGS", "-Wl,#{rpath_arg}"
 
     # Make sure the `openssl-sys` crate uses our OpenSSL.
     ENV["OPENSSL_NO_VENDOR"] = "1"
```

File: Formula/gstreamer.rb [hunk 1]
```diff
@@ -173,8 +174,10 @@ def install
     # Set `RPATH` since `cargo-c` doesn't seem to.
     # https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/issues/279
     plugin_dir = lib/"gstreamer-1.0"
-    rpath_args = [loader_path, rpath(source: plugin_dir)].map { |path| "-rpath,#{path}" }
-    ENV["RUSTFLAGS"] = "--codegen link-args=-Wl,#{rpath_args.join(",")}"
+    rpath_arg = [loader_path, rpath(source: plugin_dir)].map { |path| "-rpath,#{path}" }
+                                                        .join(",")
+    ENV["RUSTFLAGS"] = "--codegen link-args=-Wl,#{rpath_arg}"
+    ENV.append "LDFLAGS", "-Wl,#{rpath_arg}"
```
Comment (carlocab) [inline Formula/gstreamer.rb:180]: Ugh, looks like this didn't work. I guess we can drop this.

**Review by carlocab — COMMENTED** [review]
`gstreamer` is linking opportunistically with OpenCV. We can stop it with
```diff
diff --git a/Formula/gstreamer.rb b/Formula/gstreamer.rb
index b5bb27e562b..26f295b6ecc 100644
--- a/Formula/gstreamer.rb
+++ b/Formula/gstreamer.rb
@@ -151,6 +151,7 @@ class Gstreamer < Formula
       -Dgst-python:python=#{python3}
       -Dgst-plugins-bad:sctp=enabled
       -Dgst-plugins-bad:sctp-internal-usrsctp=disabled
+      -Dgst-plugins-bad:opencv=disabled
       -Dgst-plugins-good:soup=enabled
       -Dgst-plugins-rs:closedcaption=enabled
       -Dgst-plugins-rs:dav1d=enabled
```

File: Formula/gstreamer.rb [hunk 2]
```diff
@@ -149,6 +150,7 @@ def install
       -Dgst-editing-services:pygi-overrides-dir=#{site_packages}/gi/overrides
       -Dgst-python:pygi-overrides-dir=#{site_packages}/gi/overrides
       -Dgst-python:python=#{python3}
+      -Dgst-plugins-bad:opencv=disabled
```
Comment (ZhongRuoyu) [inline Formula/gstreamer.rb:153]: [`openexr` this time](https://github.com/Homebrew/homebrew-core/actions/runs/5107907041/jobs/9181323332?pr=131668#step:4:3973).

```suggestion
      -Dgst-plugins-bad:opencv=disabled
      -Dgst-plugins-bad:openexr=disabled
```

Comment (carlocab) [reply Formula/gstreamer.rb:153]: :smiling_face_with_tear:

Comment (carlocab) [reply Formula/gstreamer.rb:153]: Just add it as a dependency, `openexr` is already in the dep tree.

Comment (ZhongRuoyu) [reply Formula/gstreamer.rb:153]: Ah, sure.

File: Formula/gstreamer.rb
Comment (carlocab) [inline Formula/gstreamer.rb:1]: Also, while we're looking at `gstreamer` (but probably for a separate PR): it's total nonsense for both `nasm` and `yasm` to be build dependencies.[^1] Only one of them should be needed, I think. And likely only on Intel.

[^1]: Yes, this is [my fault](https://github.com/Homebrew/homebrew-core/commit/79c26f37b3fa09b6e2ed09ab6b179ff249e2d792). I did notice it at the time, but didn't really have the bandwidth to work out which one to drop 😅

**Review by carlocab — DISMISSED** [review]

**Comment by carlocab** [comment]
Also: it's kinda odd that a patch release needs so many revision bumps. I wonder if upstream isn't handling install names correctly.

**Review by carlocab — DISMISSED** [review]

**Review by carlocab — DISMISSED** [review]

**Review by carlocab — DISMISSED** [review]

**Review by carlocab — DISMISSED** [review]

**Comment by chenrui333** [comment]
how to proceed on this?

**Comment by carlocab** [comment]
Need to delete the uploaded bottles and then rebase.

**Comment by carlocab** [comment]
Bottles fixed, will need rebase after #132821 is merged.

**Comment by p-linnane** [comment]
3.1.9 has been released: https://github.com/AcademySoftwareFoundation/Imath/releases/tag/v3.1.9

**Comment by carlocab** [comment]
We can update if this isn't ready after CI completes.

**Review by carlocab — APPROVED** [review]

**Pull request merged**